src/overlays/actors/ovl_Toudai/ac_toudai.c

```diff
@@ -22,32 +36,265 @@ ActorProfile Toudai_Profile = {
+    var_fv0 = (frameTemp < 0) ? 1.0f : -1.0f;
+    frameTemp = ABS(frameTemp);
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Toudai/ac_toudai/aTOU_actor_dt.s")
+    if (frameTemp == 0) {
+        this->unk2C8 = 220.0f;
+        this->unk2CC = 240.0f;
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Toudai/ac_toudai/func_80AAB488_jp.s")
+    } else if (frameTemp >= 40) {
+        add_calc(&this->unk2C8, 0.0f, 0.3f, 100.0f, 1.0f);
+        add_calc(&this->unk2CC, 0.0f, 0.3f, 100.0f, 1.0f);
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Toudai/ac_toudai/func_80AAB60C_jp.s")
+    } else if (frameTemp < 10) {
+        this->unk2C8 = this->unk2C8 + (9.0f * var_fv0);
+        this->unk2CC = this->unk2CC + (14.0f * var_fv0);
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Toudai/ac_toudai/func_80AAB81C_jp.s")
+    } else if (frameTemp < 30) {
+        this->unk2C8 = this->unk2C8 + (2.5f * var_fv0);
+        this->unk2CC = this->unk2CC + (4.5f * var_fv0);
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Toudai/ac_toudai/func_80AAB840_jp.s")
+    } else {
+        this->unk2C8 = this->unk2C8 + (8.0f * var_fv0);
+        this->unk2CC = this->unk2CC + var_fv0;
+    }
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Toudai/ac_toudai/func_80AAB894_jp.s")
+    add_calc(&this->unk2D0, 255.0f, 0.1f, 100.0f, 1.0f);
+}
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Toudai/ac_toudai/func_80AAB8D8_jp.s")
+void aTOU_init(Toudai* this, Game_Play* game_play UNUSED) {
+    aTOU_setup_action(this, 1);
+}
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Toudai/ac_toudai/func_80AAB918_jp.s")
+void aTOU_wait(Toudai* this, Game_Play* game_play UNUSED) {
+    if ((common_data.time.now_sec >= 0xFD20) || (common_data.time.now_sec < 0x4650)) {
+        this->unk2BC = 1;
+        this->skeletonInfo.frameControl.speed = 1.0f;
+        aTOU_setup_action(this, 2);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Toudai/ac_toudai/func_80AAB9C8_jp.s")
+void aTOU_lighting(Toudai* this, Game_Play* game_play UNUSED) {
+    if ((common_data.time.now_sec < 64800) && (common_data.time.now_sec >= 18000)) {
+        aTOU_setup_action(this, 3);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Toudai/ac_toudai/aTOU_actor_init.s")
+void aTOU_lightout(Toudai* this, Game_Play* game_play UNUSED) {
+    if (this->unk1E8 == 0x33) {
+        this->unk2BC = 0;
+        this->skeletonInfo.frameControl.speed = 0.0f;
+        aTOU_setup_action(this, 1);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Toudai/ac_toudai/func_80AABAB8_jp.s")
+extern BaseAnimationR cKF_ba_r_obj_s_toudai;
+extern BaseAnimationR cKF_ba_r_obj_w_toudai;
+BaseAnimationR* ani[] = { &cKF_ba_r_obj_s_toudai, &cKF_ba_r_obj_w_toudai };
+ToudaiActionFunc process[] = { aTOU_init, aTOU_wait, aTOU_lighting, aTOU_lightout };
```

Comment:
process seems very global, I think it would interfere with next actors that have static process